toolkit/tools/scheduler/schedulerutils/depsolver.go

```diff
@@ -40,6 +42,21 @@ func CanSubGraph(pkgGraph *pkggraph.PkgGraph, node *pkggraph.PkgNode, useCachedI
 
 		// This node is not yet solvable
 		logger.Log.Warnf("Could not subgraph due to node: %v", pkgNode)
+
+		// If we are in trace mode, print the path from the root node to the unsolvable node
+		if logger.Log.IsLevelEnabled(logrus.TraceLevel) {
+			pt, ok := path.BellmanFordFrom(node, pkgGraph)
```

Comment:
Out of curiosity but would Bellman-Ford help us here. Since the build graph is unweighted, BFS works, or if there's no support for finding path via BFS then I think Dijkstra has similar effect/complexity on this type of graph.

Reply:
tbh it was the first agorithm that had example code in the package, I have no particular interest in the specific algorithm. I'll have a quick look to see if there is a dfs option to generate the path data but I didn't recall seeing one when I looked.

Reply:
Found a single shot shortest path alg instead of using the full BF, switched to that.

Reply:
hmm I think Yen's algorithm uses Dijkstra internally, so we can just use dijkstra to simplify it. It's also provided by gonum I think. Should help if our build graph may have up to 100k edges

Reply:
API wise the call to the Yen's function is simpler, we don't need to do the two stage process. We just get a list back. Since this is trace level I'm really not concerned about performance here. If we do this for real at debug/info level we can optimize by caching the shortest path data maybe, but I'm satisfied with yens(k=1) which simplifies down to a simple shortest path.

Reply:
sounds good to me!